azure_functions_worker/testutils.py

```diff
@@ -239,6 +247,123 @@ def _run_test(self, test, *args, **kwargs):
+                                      k=num_chars))
+
+    def is_valid_uuid(self, uuid_to_test: str, version: int = 4) -> bool:
+        """
+        Check if uuid_to_test is a valid UUID.
+        Reference: https://stackoverflow.com/a/33245493/3132415
+        """
+        try:
+            uuid_obj = uuid.UUID(uuid_to_test, version=version)
+        except ValueError:
+            return False
+        return str(uuid_obj) == uuid_to_test
+
+    def _createSharedMemoryDirectories(self, directories):
+        for temp_dir in directories:
+            temp_dir_path = os.path.join(temp_dir, consts.UNIX_TEMP_DIR_SUFFIX)
+            if not os.path.exists(temp_dir_path):
+                os.makedirs(temp_dir_path)
+
+    def _deleteSharedMemoryDirectories(self, directories):
+        for temp_dir in directories:
+            temp_dir_path = os.path.join(temp_dir, consts.UNIX_TEMP_DIR_SUFFIX)
+            shutil.rmtree(temp_dir_path)
+
+    def _setUpLinux(self):
+        self._createSharedMemoryDirectories(consts.UNIX_TEMP_DIRS)
+
+    def _tearDownLinux(self):
+        self._deleteSharedMemoryDirectories(consts.UNIX_TEMP_DIRS)
+
+    def _setUpDarwin(self):
+        """
+        Create a RAM disk on macOS.
+        Ref: https://stackoverflow.com/a/2033417/3132415
+        """
+        size_in_mb = consts.MAX_BYTES_FOR_SHARED_MEM_TRANSFER / (1024 * 1024)
+        size = 2048 * size_in_mb
+        # The following command returns the name of the created disk
+        cmd = ['hdiutil', 'attach', '-nomount', f'ram://{size}']
+        result = subprocess.run(cmd, stdout=subprocess.PIPE)
+        if result.returncode != 0:
+            raise IOError(f'Cannot create ram disk with command: {cmd} - '
+                          f'{result.stdout} - {result.stderr}')
+        disk_name = result.stdout.strip().decode()
+        # We create a volume on the disk created above and mount it
+        volume_name = 'shm'
+        cmd = ['diskutil', 'eraseVolume', 'HFS+', volume_name, disk_name]
+        result = subprocess.run(cmd, stdout=subprocess.PIPE)
+        if result.returncode != 0:
+            raise IOError(f'Cannot create volume with command: {cmd} - '
+                          f'{result.stdout} - {result.stderr}')
+        directory = f'/Volumes/{volume_name}'
+        self.created_directories = [directory]
+        self._createSharedMemoryDirectories(self.created_directories)
+        # Override the AppSetting for the duration of this test so the
+        # FileAccessorUnix can use these directories for creating memory maps
+        os.environ.update(
+            {UNIX_SHARED_MEMORY_DIRECTORIES:
+                ','.join(self.created_directories)})
+
+    def _tearDownDarwin(self):
+        volume_name = 'shm'
+        cmd = ['find', '/Volumes', '-type', 'd', '-name', volume_name,
+               '-print0', '|', 'xargs', '-0', 'umount']
+        result = subprocess.run(cmd, stdout=subprocess.PIPE)
```

Comment:
```suggestion
        cmd = "find /Volumes -type d -name 'shm*' -print0 | xargs -0 umount"
        result = subprocess.run(cmd, stdout=subprocess.PIPE, shell=True)
```

Shell=True and list->string.

Reply:
Also add `-f` flag to umount